lib/matplotlib/axes/_axes.py

```diff
@@ -3714,6 +3714,28 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
         meanprops : dict, default: None
             The style of the mean.
 
+        Notes
+        -----
+        Box plots provide insight into distribution properties of the data.
+        However, they can be challenging to interpret for the unfamiliar
+        reader. The figure below illustrates the different visual features of
+        a box plot.
```

Comment:
consider also linking to the website version so that people displaying the docstring at their terminal have somewhere to click

Reply:
There's a link to wikipedia, which IMHO is sufficient.

Furthermore, this note only gives additional context. I assume that only a minority of people will try to learn from the docstring (It's rather used for looking up signature and specific parameters). I don't want to jump extra hoops here. You are welcome to do more yourself.